addons/web/controllers/utils.py

```diff
@@ -199,6 +200,10 @@ def _get_login_redirect_url(uid, redirect=None):
     return parsed.replace(query=werkzeug.urls.url_encode(qs)).to_url()
 
 
+def is_user_internal(uid):
+    return request.env['res.users'].sudo().browse(uid).has_group('base.group_user')
+
```

Comment:
These methods are normally on `res.users` (e.g. `_is_public`, `_is_system`).

Reply:
Whether we use a `res.users`.`_is_internal` or not, is it not useful to have that utility instead of duplicating this long line (including in if statements so that multiple lines would be required)?

Reply:
xmo is not suggesting you remove that utility, he is suggesting you move it on res.users :)

Reply:
I did add a `_is_internal` in `res.users` as suggested. I'm advocating that it wouldn't be sufficient as we need that search first (when we have a user id in a login request). So, do you mean having the `is_user_internal` functionality (with sudo) on `res.user` too?

Reply:
In short: we have two methods
 * one on user (most use case), like other _is_something methods;
 * one global tool checking on uid, because we have to browse uid (before having an actual user from credentialzzz check);